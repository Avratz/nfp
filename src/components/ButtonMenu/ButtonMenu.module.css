.menuButton {
  display: block;
  padding: 0.2rem;
  position: relative;
  z-index: 100;
}

.bar {
  width: 60%;
  height: 2px;
  border-radius: 10px;
  background-color: var(--neutral-4);
  transition: all 0.25s cubic-bezier(0.15, 1, 0.7, 1);
  position: absolute;
  left: 0;
  right: 0;
  margin: 0 auto;
}

.top {
  bottom: 10px;
}

.bottom {
  top: 10px;
}

.crossTop {
  transform: rotate(-45deg);
  bottom: 15px;
  background-color: var(--neutral-1);
}

.crossBottom {
  transform: rotate(45deg);
  top: 15px;
  background-color: var(--neutral-1);
}

@media (min-width: 1200px) {
  .menuButton {
    display: none !important;
  }
}
