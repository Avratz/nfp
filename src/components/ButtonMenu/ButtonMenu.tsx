'use client'
import { Button } from '@/ui/Button/Button'

import styles from './ButtonMenu.module.css'
import { cx } from 'class-variance-authority'

export function ButtonMenu({
  isOpen,
  setIsOpen,
}: {
  isOpen: boolean
  setIsOpen: (isOpen: boolean) => void
}) {
  function handleClick() {
    setIsOpen(!isOpen)
  }

  return (
    <Button
      intent="icon"
      className={styles.menuButton}
      onClick={handleClick}
      title="Menu"
    >
      <div
        className={cx(styles.bar, styles.top, isOpen ? styles.crossTop : '')}
      ></div>
      <div
        className={cx(
          styles.bar,
          styles.bottom,
          isOpen ? styles.crossBottom : '',
        )}
      ></div>
    </Button>
  )
}
