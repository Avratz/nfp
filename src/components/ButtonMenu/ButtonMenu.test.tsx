import { render } from '@testing-library/react'
import { ButtonMenu } from './ButtonMenu'

describe('ButtonMenu', () => {
  it('renders ButtonMenu closed', () => {
    const { container } = render(
      <ButtonMenu isOpen={false} setIsOpen={() => null} />,
    )
    expect(container).toMatchSnapshot()
  })

  it('renders ButtonMenu open', () => {
    const { container } = render(
      <ButtonMenu isOpen={true} setIsOpen={() => null} />,
    )
    expect(container).toMatchSnapshot()
  })
})
