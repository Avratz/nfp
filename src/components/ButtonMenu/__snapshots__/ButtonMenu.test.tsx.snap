// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonMenu renders ButtonMenu closed 1`] = `
<div>
  <button
    class="button icon base menuButton"
    title="Menu"
  >
    <div
      class="bar top"
    />
    <div
      class="bar bottom"
    />
  </button>
</div>
`;

exports[`ButtonMenu renders ButtonMenu open 1`] = `
<div>
  <button
    class="button icon base menuButton"
    title="Menu"
  >
    <div
      class="bar top crossTop"
    />
    <div
      class="bar bottom crossBottom"
    />
  </button>
</div>
`;
